### PR TITLE
Feat/15086/add banner severities component

### LINKED ISSUE #15086: [Ext Nav] Create component: Banner
### Description
Create a reusable UI component for AvatarWithToken

### References
[Figma Link](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/Design-System-2.0?node-id=1915%3A12258)
UI component guidelines: TBC
CSS guidelines: TBC
Testing guidelines: TBC

### Files needed
- `index.js`
- `index.scss`
- `Banner.js`
- `Banner.stories.js`
- `Banner.test.js`
- `README.mdx`

### Technical details
TBC

### Acceptance criteria
- [ ] Uses color, typography, shadows design tokens 
- [ ] Uses semantic html
- [ ] PropTypes have descriptions
- [ ] Has storybook story with controls
- [ ] Has documentation in MDX
- [ ] Has unit tests and 90% coverage
- [ ] Works in Chrome and Firefox
- [ ] Performance tested: no unnecessary re-renders or other performance concerns

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

*Fixes: #15086 

### Before
Screenshot 2023-01-19 at 4 04 52 PM

### After
Fixed `BannerBase` to be full width/block
Screenshot 2023-01-19 at 4 04 25 PM

Screenshot 2023-01-19 at 4 05 39 PM

## Manual Testing Steps

`yarn jest ui/components/component-library/banner/banner.test.js`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/banner/banner.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      default:
+        return (
+          <Icon
+            name={ICON_NAMES.INFO_FILLED}
+            size={SIZES.LG}
+            color={COLORS.PRIMARY_DEFAULT}
+          />
+        );
+    }
+  };
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner/banner.js:56]: Could you save some lines of code here by returning an object and spreading it to the Icon component? Something like:

```jsx
  const severityIcon = () => {
    switch (severity) {
      case SEVERITIES.DANGER:
        return {
          name: ICON_NAMES.DANGER_FILLED,
          color: COLORS.ERROR_DEFAULT,
        };
      case SEVERITIES.WARNING:
        return {
          name: ICON_NAMES.WARNING_FILLED,
          color: COLORS.WARNING_DEFAULT,
        };
      case SEVERITIES.SUCCESS:
        return {
          name: ICON_NAMES.CHECK_CIRCLE_ON_FILLED,
          color: COLORS.SUCCESS_DEFAULT,
        };
      // Defaults to SEVERITIES.INFO
      default:
        return {
          name: ICON_NAMES.INFO_FILLED,
          color: COLORS.PRIMARY_DEFAULT,
        };
    }
  };

return (
    <BannerBase
      startAccessory={<Icon size={SIZES.LG} {...severityIcon()} />}
      backgroundColor={severityBackground()}
```

```

File: ui/components/component-library/banner/banner.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  );
+};
+
+Banner.propTypes = {
+  /**
+   * An additional className to apply to the Banner
+   */
+  className: PropTypes.string,
+  /**
+   * TODO WRITE HERE
```
Comment (georgewrmarshall) [inline ui/components/component-library/banner/banner.js:97]: Update to do
```suggestion
   * Use the `severity` prop and the `SEVERITIES` object from `./ui/helpers/constants/design-system.js` to change the context of `Banner`.
   * Possible options: `SEVERITIES.INFO`(Default), `SEVERITIES.WARNING`, `SEVERITIES.DANGER`, `SEVERITIES.SUCCESS`
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking great! Some minor suggestions

File: ui/components/component-library/banner/README.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+The `Banner` component is used for inline notifcations and has different colors based on context's severity.
+
+<Canvas>
+  <Story id="components-componentlibrary-banner--default-story" />
+</Canvas>
+
+## Props
+
+The `Banner` accepts all props below as well as all [Box](/docs/ui-components-ui-box-box-stories-js--default-story#props) component props
```
Comment (NidhiKJha) [inline ui/components/component-library/banner/README.mdx:15]: Need to update the box url 
```suggestion
The `Banner` accepts all props below as well as all [Box](/docs/components-ui-box--default-story#props) component props
```

File: ui/components/component-library/banner/README.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  {`Description shouldn't repeat title. 1-3 lines. Can contain a `}
+  <ButtonLink size={SIZES.AUTO} href="https://metamask.io/" target="_blank">
+    hyperlink.
+  </ButtonLink>
+</Banner>;
+```
+
+### Action Button Label, onClick, & Props
+
+Use the `actionButtonLabel` prop to pass text, `actionButtonOnClick` prop to pass an onClick handler, and `actionButtonProps` prop to pass an object of [ButtonLink props](/docs/ui-components-component-library-button-link-button-link-stories-js--default-story) for the action
```
Comment (NidhiKJha) [inline ui/components/component-library/banner/README.mdx:96]:
```suggestion
Use the `actionButtonLabel` prop to pass text, `actionButtonOnClick` prop to pass an onClick handler, and `actionButtonProps` prop to pass an object of [ButtonLink props](/docs/components-componentlibrary-buttonlink--default-story) for the action
```

**Review by NidhiKJha — DISMISSED** [review]
Looks really good. Just a couple of URL update in Readme

**Review by darkwing — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

We have one accessibility violation but I think it's because the background color isn't using our `background-default` `#FFFFF`

Screenshot 2023-01-25 at 9 47 23 AM

Please also connect with @Akatori-Design because we have some inconsistencies with our code component and Figma component.

**Pull request merged**